src/ConnectedMode/UI/ManageConnections/ManageConnectionsDialog.xaml.cs

```diff
@@ -52,7 +52,8 @@ private void EditConnection_Clicked(object sender, RoutedEventArgs e)
 
         private void NewConnection_Clicked(object sender, RoutedEventArgs e)
         {
-            if (GetTransientConnection() is {} transientConnection && CredentialsDialogSucceeded(transientConnection) && GetCompleteConnection(transientConnection) is {} completeConnection)
+            if (GetTransientConnection() is {} partialConnection && GetCredentialsDialog(partialConnection) is {} credentialsDialog && 
```

Comment:
1) formatting is incorrect here. 1 part of boolean expression per line, line breaks before operators.
2) GetCredentialsDialog can never return null, so the condition feels a bit hacky. At this point I would convert this `if (a && b && c && d)` into
```
if(!a) return;
var dialog = b();
if(!c) return;
if(connection.Type == SonarCloud && !d) return;
viewModel.add()
```